peps/pep-0781.rst

```diff
@@ -0,0 +1,120 @@
+PEP: 781
+Title: Make ``TYPE_CHECKING`` builtin
+Author: Inada Naoki <songofacandy@gmail.com>
+Discussions-To: https://discuss.python.org/t/85728
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Created: 24-Mar-2025
+Python-Version: 3.14
+Post-History: `11-Jan-2025 <https://discuss.python.org/t/76766>`__,
+              `24-Mar-2025 <https://discuss.python.org/t/85728>`__,
+
+
+Abstract
+========
+
+This PEP proposes adding a new builtin, :data:`!TYPE_CHECKING`, to improve
+the experience of writing Python code with type annotations. It is evaluated
+as ``True`` when the code is being analyzed by a static type checker, and as
+``False`` during normal runtime execution. Unlike :data:`typing.TYPE_CHECKING`,
+which this keyword replaces, it does not require an import statement.
+
+
+Motivation
+==========
+
+Type annotations were defined for Python by :pep:`484`, and have enjoyed
+widespread adoption. A challenge with fully-annotated code is that many
+more imports are required in order to bring the relevant name into scope,
+potentially causing import cycles without careful design. This has been
+recognized by :pep:`563` and later :pep:`649`, which introduce two different
+mechanisms for deferred evaluation of type annotations. As :pep:`563` notes,
```

Comment:
PEP 563 was linked in the previous sentence, so stylistically it might be preferable to avoid this link?
```suggestion
mechanisms for deferred evaluation of type annotations. As PEP 563 notes,
```